### PR TITLE
Move the ground truth closure computation to C++

### PR DESCRIPTION
**Pull request by saurabh1002**
1. Move GT closures computation to C++ and make it fast, python equivalent takes ages for large datasets:
Nothing changes in the way we compute gt closures, except for the bug fixes in VoxelHashMap from KISS-ICP which was internally used to compute the overlap score between two local segments of the trajectory

2. Provide a standalone `gt_closure_pipeline` to generate and store ground truth closures:
Useful since some of the other approaches for loop closures have started to use a similar idea for gt closures as proposed in our paper

3. Performance upgrade is from hours/days to minutes -> C++ to the win
That means our approach can be evaluated on more datasets faster.

### CODE REVIEW

File: cpp/gt_closures/GTClosures.cpp
Comment (tizianoGuadagnino) [inline cpp/gt_closures/GTClosures.cpp:1]: Put license please

**Comment by benemer** [comment]
@saurabh1002 I marked this as draft because it looks like you are still working on it. Just let us know when it's ready to review :+1:

**Review by benemer — CHANGES_REQUESTED** [review]

**Comment by benemer** [comment]
Amazing speed-up, love it :)

**Review by benemer — APPROVED** [review]

File: cpp/gt_closures/GTClosures.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+#include <vector>
+
+#include "VoxelHashSet.hpp"
+
+namespace gt_closures {
+
+GTClosures::GTClosures(const int dataset_size,
+                       const double sampling_distance,
+                       const double overlap_threshold,
+                       double voxel_size,
```
Comment (tizianoGuadagnino) [inline cpp/gt_closures/GTClosures.cpp:43]:
```suggestion
                       const double voxel_size,
```

**Pull request merged**